p2p/chain.py

```diff
@@ -281,16 +287,31 @@ def register_peer(self, peer: BasePeer) -> None:
                 pending_replies = request_func(missing)
                 continue
 
-            parts.extend(received)
+            received_keys_by_peer = groupby(operator.attrgetter('peer'), received)
```

Comment:
All items in `received` will have the same 'peer' here. That's why in https://github.com/ethereum/py-evm/pull/779 I used a tuple (`(peer, items)`) instead of adding a peer attribute to `DownloadedBlockPart`

Reply:
I think this code would be significantly simpler if you did the same

Reply:
👍 I'll rebase on top of that.